src/Stratis.Bitcoin/Connection/PeerBanning.cs

```diff
@@ -131,7 +83,14 @@ public void BanPeer(IPEndPoint endpoint, int banTimeSeconds, string reason = nul
             if (banPeer)
             {
                 this.logger.LogDebug("Peer '{0}' banned for reason '{1}'.", endpoint, reason);
-                this.banStore.BanPeer(endpoint, this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds));
+
+                PeerAddress peerToAdd = PeerAddress.Create(endpoint, endpoint.Address);
+
+                peerToAdd.IsBanned = true;
+                peerToAdd.BannedReason = reason;
+                peerToAdd.BanUntil = this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds);
+
+                this.peerAddressManager.Peers.TryAdd(peerToAdd.Endpoint, peerToAdd);
```

Comment:
Its probably not such a good idea to have the `Peers` dictionary as public this means any outside component can modify/add peers, this is effectively the store so perhaps we should protect it (maybe expose the values only) @fassadlr any reason its a public property?

Adding to this list has to be done in one place only I would suggest to change the method on IPeerAddressManager.Add to return the newly created PeerAddress so the caller can then set the BannedUntil property

Reply:
I just made this change in another PR

Reply:
Can you send over a link or commit details and I'll take a look, cheers.

Reply:
yeah #1204

Reply:
Ta

Reply:
Merged in #1204 to my branch and changed the code for this issue accordingly.  Looks a lot cleaner now.  Will create a new PR.  Cheers.